onnxruntime/contrib_ops/rocm/diffusion/group_norm_ck.cuh

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef USE_COMPOSABLE_KERNEL
+#include "ck/ck.hpp"
+#include "ck/tensor_operation/gpu/device/tensor_layout.hpp"
+#include "ck/tensor_operation/gpu/element/element_wise_operation.hpp"
+#include "contrib_ops/rocm/diffusion/group_norm_ck_impl/impl.cuh"
+#endif  // USE_COMPOSABLE_KERNEL
+
+#include "contrib_ops/rocm/diffusion/group_norm_common.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+#ifdef USE_COMPOSABLE_KERNEL
+
+template <typename T>
+struct DataTypeAdaptor {
```

Comment:
Call for refactor in latter PRs, we are duplicating too many of this adaptor 🤦‍♂️.